kabuki.py

```diff
@@ -0,0 +1,235 @@
+#!/usr/bin/env python
+# Main kabuki.py
+# imports - use the rpi-rgb-led-matrix instructions to install dependencies
+import queue
+import time
+from multiprocessing import JoinableQueue
+from multiprocessing import Process
+
+from PIL import Image
+from PIL import ImageDraw
+from rgbmatrix import graphics
+
+from face import Face
+
+# constants used throughout project
+WIDTH = 32
+HEIGHT = 32
+SPEED = 1.0 / 20.0
+IDLE_TIME = 50
+EYES = 0
+MOUTH = 16
+
+
+class Kabuki:
+    def __init__(self, matrix):
+        self.matrix = matrix
+        self.face = Face()
+        self.eye_queue = JoinableQueue()
+        self.mouth_queue = JoinableQueue()
+        self.render_queue = JoinableQueue()
+        self.loop_proc = Process(target=self.loop)
+        self.compute_proc = Process(target=self.compute_frame)
+        self.render_proc = Process(target=self.render_loop)
+        self.reset_face()
+
+    def start_render(self):
+        self.loop_proc.start()
+        # self.loop_proc.join()
+        self.compute_proc.start()
+        self.render_proc.start()
+        # self.compute_proc.join()
+        # self.render_proc.join()

```

Comment:
If you follow the above recommendation, you can join all three at the end but since they're all infinite loops nothing will happen.